packages/dev/core/src/Engines/Native/nativePipelineContext.ts

```diff
@@ -4,16 +4,27 @@ import type { IMatrixLike, IVector2Like, IVector3Like, IVector4Like, IColor3Like
 import type { IPipelineContext } from "../IPipelineContext";
 import type { NativeEngine } from "../nativeEngine";
 
+export enum CompilationState {
+    uncompiled,
+    compiled,
+    error,
+}

```

Comment:
Enums are expensive in terms of TS code generation. See [here](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBMAdgVwLZwMIVWAlgGwEMZcJEBlGY4OAbwFgAoOFuZRAY2z32ABMANE1ZwuOAvyHNWwKFGhSAvkA). It seems like you can get away with a boolean and an Error object?

Reply:
Okay, I changed it.